### ISSUE TITLE
Store the static and dynamic baseSite data to the baseSite service

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#9724]
The current baseService does not expose the base site data that has been captured during a multi-site setup. This means that if we want to drive Spartacus features based on baseSite data, we need to another backend call. This is bad for performance. 

Adding the base site data from the multi-site setup is relative complex as the logic to resolve the data must be done prior to anything in Spartacus. Hence some of the capabilities have not been initialised at that stage. While this remain true, we like to make this transparent to the consumers of the BaseSiteService, so that the active base site data can be observed without knowing how this data has been captured. 

In order to do so, the `OccConfigLoaderService` must store its results in the ngrx store.

**Comment by Platonn** [comment]
As a result of this ticket it would be good to drive theme and channel (b2b/b2c) dynamically. And thanks to that dynamically provide configuration for a storefront version b2b v.s b2c (i.e. in Spartacus' dynamic `ConfigInitializer`).

**Comment by tobi-or-not-tobi** [comment]
> As a result of this ticket it would be good to drive theme and channel (b2b/b2c) dynamically. And thanks to that dynamically provide configuration for a storefront version b2b v.s b2c (i.e. in Spartacus' dynamic `ConfigInitializer`).

We'd also like to use the active site uid to load static resources, such as a site specific sw config, logo, icons, etc.

**Comment by Platonn** [comment]
I think not whole scope of the ticket https://github.com/SAP/spartacus/issues/8459 was covered by its PR. So we may want to cover it in this ticket or a related one... Not sure now.

**Comment by Platonn** [comment]
Closing in favour of https://github.com/SAP/spartacus/issues/11040
